### PR TITLE
Elevation of orbit wrt a location on attractor

### PR DESCRIPTION
**Pull request by Yash-10**
Adapted implementation from #1299 since it was particularly difficult to validate against Orekit (see poliastro/validation#35). It requires LST as input (longitude would have been ok too, but would make the elevation function more complex).

Instead, I tried to add a utility function that returns LST given coordinates of some location on Earth. Since that utility function was focused on Earth (because it uses Greenwich mean sidereal time), it was added in earth utils.

Still polishing tiny bits...

- [x] Add test for `get_local_sidereal_time`

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Read the Docs failure is a bit difficult to see, but here it is:

```
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/events/index.rst:64: ERROR: Unknown target name: "u".
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/events/index.rst:64: ERROR: Unknown target name: "u".
```

**Comment by Yash-10** [comment]
It's hard for me to inspect the doc build locally since I am faced with an old error which we discussed in #1303 but still remains a mystery...

(That error comes up even after a clean install in a new environment)

**Comment by Yash-10** [comment]
For some reason, the documentation build no longer occurs! It's ready for review.

File: src/poliastro/core/events.py [hunk 1]
```diff
@@ -84,3 +84,55 @@ def line_of_sight(r1, r2, R):
...
+        Polar radius of the central attractor.
+    H: float
+        Elevation, above the ellipsoidal surface.
+
+    """
+    ecc = np.sqrt(1 - (R_p / R) ** 2)
+    denom = np.sqrt(1 - ecc**2 * np.sin(phi) ** 2)
+    g1 = H + (R / denom)
+    g2 = H + (1 - ecc**2) * R / denom
+    station_coords = np.array(
```
Comment (astrojuanlu) [inline src/poliastro/core/events.py:116]: Wondering if we have these conversion functions somewhere already?

File: tests/tests_twobody/test_orbit.py [hunk 2]
```diff
@@ -1274,3 +1274,45 @@ def test_propagation_near_parabolic_orbits_does_not_hang(near_parabolic):
     assert_quantity_allclose(
         (orb_final.epoch - near_parabolic.epoch).to(u.s), near_parabolic.period
     )
+
+
+def test_orbit_elevation():
+    # From example 5.9 Howard Curtis fourth edition.
+    r = np.array([-2032.4, 4591.2, -4544.8]) << u.km
```
Comment (astrojuanlu) [inline tests/tests_twobody/test_orbit.py:1281]: This also works:

```suggestion
    r = [-2032.4, 4591.2, -4544.8] << u.km
```

**Review by astrojuanlu — COMMENTED** [review]
Did a quick, superficial review, thanks @Yash-10 !

File: src/poliastro/core/events.py (hunk 1 above)
Comment (Yash-10) [reply src/poliastro/core/events.py]: That line performs a transformation between geocentric equatorial coordinates to the topocentric horizon coordinate system.

Using `astropy`, we could define [`AltAz`](https://docs.astropy.org/en/stable/api/astropy.coordinates.AltAz.html) and use something like `transform_to` to do the transformation.

As decided, keeping performance in mind, we could add our own implementations for that.

**Comment by Yash-10** [comment]
@astrojuanlu I wonder if you had time to look at this again?

Thanks!

**Comment by Yash-10** [comment]
@astrojuanlu can we take some decision on this PR?

(I don't know the exact cause of the errors, it seems it needs some investigation).

**Comment by astrojuanlu** [comment]
We'll try to get this merged or otherwise state what's needed for it to happen.

**Comment by astrojuanlu** [comment]
Well well well! All the tests are passing! 😃 

@Yash-10 do you want to have a look at the documentation failures?

```
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/events/index.rst:64: ERROR: Unknown target name: "u".
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/events/index.rst:64: ERROR: Unknown target name: "u".
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/util/index.rst:55: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/core/util/index.rst:60: WARNING: Literal block ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/poliastro/checkouts/1486/docs/source/autoapi/poliastro/twobody/orbit/scalar/index.rst:392: WARNING: Definition list ends without a blank line; unexpected unindent.
```

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Possible fixes

**Comment by Yash-10** [comment]
Thanks, @astrojuanlu! Indeed, I didn't fix my docstrings. Now the tests seem to be fine.

**Review by astrojuanlu — APPROVED** [review]
This is good to merge!

**Comment by astrojuanlu** [comment]
Thanks for your patience @Yash-10 🙏🏽

**Pull request merged**